### PR TITLE
Python django upgrade

### PR DESCRIPTION
**Pull request by dpwrussell**
Added support for Python3
Added support for Django >= 1.11
Added support for OMERO 5.6
Black code formatting
Bump autotag version to 3.1.0
Bump tagsearch version to 3.1.0

### CODE REVIEW

File: autotag/setup.py [hunk 1]
```diff
@@ -64,68 +62,65 @@ def finalize_options(self):
...
+        "Development Status :: 5 - Production/Stable",
+        "Environment :: Web Environment",
+        "Framework :: Django",
+        "Intended Audience :: End Users/Desktop",
+        "Intended Audience :: Science/Research",
+        "License :: OSI Approved :: GNU Affero General Public License v3",
+        "Natural Language :: English",
+        "Operating System :: OS Independent",
+        "Programming Language :: JavaScript",
+        "Programming Language :: Python :: 2",
```
Comment (mtbc) [inline autotag/setup.py]:
```suggestion
        "Programming Language :: Python :: 3",
```

**Comment by will-moore** [comment]
Basic functional testing of autotag and tagsearch is looking good.
Creating, adding, removing tags in auto-tag and then searching for a hierarchy of these tags in tagsearch is 👍 .

**Comment by will-moore** [comment]
Quick browse of code changes looks fine...
In other repos, we have done ```install_requires=['omero-web>=5.6.0'],``` in ```setup.py``` but not a blocker.

File: tagsearch/setup.py [hunk 2]
```diff
@@ -11,51 +11,50 @@ def read_file(fname, content_type=None):
...
+        "Development Status :: 5 - Production/Stable",
+        "Environment :: Web Environment",
+        "Framework :: Django",
+        "Intended Audience :: End Users/Desktop",
+        "Intended Audience :: Science/Research",
+        "License :: OSI Approved :: GNU Affero General Public License v3",
+        "Natural Language :: English",
+        "Operating System :: OS Independent",
+        "Programming Language :: JavaScript",
+        "Programming Language :: Python :: 2",
```
[the remark this answers is not in the document]
Comment (dpwrussell) [reply tagsearch/setup.py]: @mtbc I've deliverated about this as technically, it is also compatible with Python 2, but given that it's end of life, I agree and will incorporate that change.

**Comment by dpwrussell** [comment]
> Quick browse of code changes looks fine...
> In other repos, we have done `install_requires=['omero-web>=5.6.0'],` in `setup.py` but not a blocker.

@will-moore Thanks for the review! Are you putting in this requirement even if it is backwards compatible with <5.6.0? Very happy to put it in if that's in line with what everything else is doing.

**Comment by will-moore** [comment]
@dpwrussell yes, that's the decision we made (not to be backwards compatible), partly to simplify our lives and also to encourage everyone to upgrade to OMERO 5.6.0. But not essential for you to follow suit.

**Comment by will-moore** [comment]
I've tried installing the webtagging apps directly from this branch on our ci,
```
pip install "git+git://github.com/dpwrussell/webtagging.git@python_django_upgrade#egg=omero-webtagging-autotag&subdirectory=autotag"
omero config append omero.web.apps '"omero_webtagging_autotag"'
omero config append omero.web.ui.center_plugins '["Auto Tag", "omero_webtagging_autotag/auto_tag_init.js.html", "auto_tag_panel"]'

# OMERO.webtagging-tagsearch installation
# pip install omero-webtagging-tagsearch
pip install "git+git://github.com/dpwrussell/webtagging.git@python_django_upgrade#egg=omero-webtagging-tagsearch&subdirectory=tagsearch"
omero config append omero.web.apps '"omero_webtagging_tagsearch"'
omero config append omero.web.ui.top_links '["Tag Search", "tagsearch"]'
```

This is working *except* for the building of the ```bundle.js```.
We saw similar issues on py3 with ```omero-parade``` and ```omero-figure``` where the npm step wasn't happening. Fixed with https://github.com/ome/omero-parade/pull/71 and https://github.com/ome/omero-figure/pull/359.

Before these fixes, the apps were still installable from a release/tag where the js bundle has already been built and pushed to pypi. So this is not a blocker for release - it just prevents testing of merge branches via CI.

Probably the easiest option is to merge this PR and release as it is. Then we can test the install of that release (which we expect to be working). If not, then we'll need extra fixes and release, but let's see...

Then we can try to apply the npm build fixes later as needed?

Does that sound OK?

**Comment by dpwrussell** [comment]
> @dpwrussell yes, that's the decision we made (not to be backwards compatible), partly to simplify our lives and also to encourage everyone to upgrade to OMERO 5.6.0. But not essential for you to follow suit.

I also decided that it's best to encourage upgrade so will add:
```python
install_requires=['omero-web>=5.6.0'],
python_requires='~=3.6',
```

**Comment by will-moore** [comment]
I think it would be good to support python 3.5 too, since we will try to support that soon on various platforms.

**Comment by dpwrussell** [comment]
> This is working _except_ for the building of the `bundle.js`.
> We saw similar issues on py3 with `omero-parade` and `omero-figure` where the npm step wasn't happening. Fixed with [ome/omero-parade#71](https://github.com/ome/omero-parade/pull/71) and [ome/omero-figure#359](https://github.com/ome/omero-figure/pull/359).

I went ahead and fixed the `sdist`. Thanks for the links.

I just need to update the README and then I think this is ready to merge and release.

Thanks @will-moore and @mtbc

**Comment by dpwrussell** [comment]
> I think it would be good to support python 3.5 too, since we will try to support that soon on various platforms.

Shouldn't be a problem.

**Comment by dpwrussell** [comment]
@will-moore Did you want to do any further testing (now that I've fixed the npm build part of setup.py) before I do the release?

**Comment by will-moore** [comment]
Install from branch as above looks like it's working fine.
Please go ahead with the release,
 Thanks!

**Pull request merged**